src/extensions/lib/src/add-credits.ts

```diff
@@ -7,14 +7,49 @@ export const SOURCE_ID_TO_TARGET_ID: Record<string, string> = {
 	203478287: '163146',
 };
 
+type CreditsAddition = {
+	meta: { amountInDollars?: number };
+};
+
 type AddCreditsData = {
 	github_id: string;
 	amount: number;
 	reason: 'recurring_sponsorship' | 'one_time_sponsorship' | 'tier_changed';
 	meta: { amountInDollars: number };
 };
 
-export const addCredits = async ({ github_id, amount, reason, meta }: AddCreditsData, { services, database, getSchema, env }: ApiExtensionContext) => {
+export const getUserBonus = async (githubId: string | null, incomingAmountInDollars: number, { services, database, getSchema, env }: ApiExtensionContext) => {
+	const { ItemsService } = services;
+
+	if (!env.CREDITS_BONUS_PER_100_DOLLARS || !env.MAX_CREDITS_BONUS) {
+		throw new Error('CREDITS_BONUS_PER_100_DOLLARS or MAX_CREDITS_BONUS was not provided');
+	}
+
+	const maxCreditsBonus = parseInt(env.MAX_CREDITS_BONUS, 10);
+
+	const creditsAdditionsService = new ItemsService('gp_credits_additions', {
+		schema: await getSchema({ database }),
+		knex: database,
+	});
+
+	const additionsInLastYear = await creditsAdditionsService.readByQuery({
+		filter: {
+			github_id: { _eq: githubId },
+			reason: { _in: [ 'recurring_sponsorship', 'one_time_sponsorship', 'tier_changed' ] },
+			date_created: { _gte: new Date(Date.now() - 365 * 24 * 60 * 60 * 1000) },
```

Comment:
Let's change this to 367 days to cover leap years and intra-day hour differences.